temporalio/workflow.py

```diff
@@ -906,6 +906,8 @@ def upsert_search_attributes(
             DEPRECATED and if used, result in invalid key types on the
             typed_search_attributes property in the info.
     """
+    if not attributes:
```

Comment:
A type checker should disallow this. Note, many of our calls will fail if you don't pass in the types the parameter says it accepts, not just this one. I am unsure it is reasonable to add runtime parameter type checking to all functions, but maybe it can be a clearer error message?

If we do want to make this function accept `Optional` `attributes`, we should alter the signature of the method I think. But I don't think we want to accept `None` here. Do we want to at least raise a clearer error message instead of blindly accepting this bad input?